### PR TITLE
Feat/homepage to functions

### PR DESCRIPTION
**Pull request by Ellenn-A**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [ ] Bug Fix
- [x] Feature
- [ ] Documentation Update
- [ ] Code style update (formatting, local variables)
- [ ] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets
- https://digicatapult.atlassian.net/browse/VR-146

## High level description

A homepage to all the Functions / Applications that Veritable offers.

## Detailed description

A homepage with different cards to navigate to different functions/applications in veritable. The homepage button in the navigation also navigates to the homepage. 

## Describe alternatives you've considered

N/A

## Operational impact

Allows the user to view all the functionality veritable has to offer and to navigate to it from a single place. 

## Additional context

N/A

### CODE REVIEW

**Review by n3op2 — COMMENTED** [review]
Yay, a new button enabled! But categories should be the home page?

File: src/controllers/categories/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import HomepageTemplates from '../../views/homepage.js'
+import { HTML, HTMLController } from '../HTMLController.js'
+
+@singleton()
+@injectable()
+@Security('oauth2')
+@Route('/')
+@Produces('text/html')
+export class HomepageController extends HTMLController {
```
Comment (n3op2) [inline src/controllers/categories/index.ts:14]: just wonder why HomepageController is under `categories` folder?

Comment (mattdean-digicatapult) [reply src/controllers/categories/index.ts:14]: agreed I think this should just be in the controllers directory as `homepageController.ts` or similar

File: src/views/homepage.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return (
+      <Page title="Veritable - Homepage" heading="Homepage" headerLinks={[{ name: '', url: '/' }]}>
+        <div
+          class="main connections"
+          hx-get="/"
+          hx-trigger="every 10s"
+          hx-select="#search-results"
+          hx-target="#search-results"
+          hx-swap="outerHTML"
+          hx-include="#connection-search-input"
```
Comment (mattdean-digicatapult) [inline src/views/homepage.tsx:18]: why are we polling here?

File: src/views/homepage.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        ></div>
+        <div class="center-category">
+          <div class="category-container">
+            <div class="category-item" onclick="location.href='#';">
```
Comment (mattdean-digicatapult) [inline src/views/homepage.tsx:22]: Should one of these not take us to the connections page? Can you ask Esther please

File: public/styles/main.css [hunk 4]
```diff
@@ -333,6 +333,54 @@ a.connections-table.icon {
...
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-template-rows: 1fr 1fr;
+  gap: 25px;
+  width: 70%;
+  height: 40rem;
+  cursor: pointer;
+}
+
+.category-item {
```
Comment (mattdean-digicatapult) [inline public/styles/main.css:346]: you probably want min and max sizes on these

File: public/styles/main.css [hunk 5]
```diff
@@ -333,6 +333,54 @@ a.connections-table.icon {
   width: 12px;
   background-image: url('../images/info-accent.svg');
 }
+.category-container {
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-template-rows: 1fr 1fr;
```
Comment (mattdean-digicatapult) [inline public/styles/main.css:339]: I don't think we need to force a 2x2 grid here. Have a look at `repeat` and `auto-fit` in particular

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]

**Pull request merged**